test-data/unit/check-unreachable-code.test

```diff
@@ -715,3 +715,179 @@ if sys.version_info[0] >= 2:
     reveal_type('')  # N: Revealed type is 'builtins.str'
 reveal_type('')  # N: Revealed type is 'builtins.str'
 [builtins fixtures/ops.pyi]
+
+[case testUnreachableFlagWithBadControlFlow]
+# flags: --disallow-inferred-unreachable
+a: int
+if isinstance(a, int):
```

Comment:
Add test with a function that accepts a type variable with value restriction and performs an `isinstance` check that is always true/false for some value of the type variable. It may be reasonable to disable unreachability checks in a function with a type variable restriction (at least for now), since the correct implementation may be tricky and such type variables are relatively rare.